ext/intl/idn/idn.c

```diff
@@ -226,6 +226,9 @@ static void php_intl_idn_to(INTERNAL_FUNCTION_PARAMETERS,
 #if (ZEND_GCC_VERSION >= 4002) && !defined(__clang__)
 # pragma GCC diagnostic push
 # pragma GCC diagnostic ignored "-Wdeprecated-declarations"
+#elif defined(__clang__)
+# pragma clang diagnostic push
+# pragma clang diagnostic ignored "-Wdeprecated-declarations"
```

Comment:
Possibly invert the check order so you don't have to check for `defined(__clang__)` twice?

Reply:
Yes, makes sense. However, I guess we might need a version check as well, but for which version?

Reply:
I don't know -- I think we can just add one once someone complains :)

Reply:
Okay, makes sense.